python/ray/dataframe/dataframe.py

```diff
@@ -35,18 +35,21 @@
     _blocks_to_col,
     _blocks_to_row,
     _create_block_partitions,
+    _mask_block_partitions,
     _inherit_docstrings,
     _reindex_helper)
 from . import get_npartitions
 from .index_metadata import _IndexMetadata
+from .indexing import _Loc_Indexer, _iLoc_Indexer
 
 
 @_inherit_docstrings(pd.DataFrame)
 class DataFrame(object):
 
     def __init__(self, data=None, index=None, columns=None, dtype=None,
                  copy=False, col_partitions=None, row_partitions=None,
-                 block_partitions=None, row_metadata=None, col_metadata=None):
+                 block_partitions=None, row_metadata=None, col_metadata=None,
+                 partial=False):
```

Comment:
Would it make more sense to have a `DataFrameView` object as a subclass of this one?

Reply:
As I'm reading through this, I think it might make more sense to have it as a subclass. Changing the way that `_block_partitions` is handled in the case that it's a view makes complicated code more complicated. 

Alternatively, if we decide otherwise, I would still like to see more comments about the `_block_partitions` changes.

Reply:
I agree we should have it as a subclass. `DataFrameView` object make sense. 

I'm refactoring parts of my `indexing.py` to make enlargement inside the parent class of the `_LocIndexer` and `_iLocIndexer`. So that we have cleaner code in `indexing.py`